iOS/DuckDuckGo/DefaultBrowserPrompt/DefaultBrowserPromptUserTypeManager.swift

```
@@ -0,0 +1,97 @@
+//
+//  DefaultBrowserPromptUserTypeManager.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import BrowserServicesKit
+import Persistence
+import SetDefaultBrowserCore
+
+final class DefaultBrowserPromptUserTypeManager {
+    private let store: DefaultBrowserPromptUserTypeStoring
+    private let statisticsStore: StatisticsStore
+
+    init(store: DefaultBrowserPromptUserTypeStoring, statisticsStore: StatisticsStore = StatisticsUserDefaults()) {
+        self.store = store
+        self.statisticsStore = statisticsStore
+    }
+
+    func persistUserType() {
+        // Save value only on first launch.
+        guard store.userType() == nil else { return }
+
+        let userType: DefaultBrowserPromptUserType = if statisticsStore.hasInstallStatistics {
+            .existing
+        } else if statisticsStore.variant == VariantIOS.returningUser.name {
+            .returning
+        } else {
+            .new
+        }
+        store.save(userType: userType)
+    }
+}
+
+// MARK: - DefaultBrowserPromptUserTypeProviding
+
+extension DefaultBrowserPromptUserTypeManager: DefaultBrowserPromptUserTypeProviding {
+
+    func currentUserType() -> DefaultBrowserPromptUserType {
+        store.userType() ?? .new
```

Now that we use a throwing storage instead of UserDefaults there are situation where saving or retrieving may fail. It should not happen but it can happen. 

Currently I’m defaulting the user type to `.new` but we could change to return an optional and if that happens we don’t show the prompt for those users. This should be an edge case.

---

I'd probably go for the safer option of doing nothing. Consider firing a pixel / generating an event here.